api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -268,11 +269,20 @@ public User createUser(
     if (demographicSurvey != null) demographicSurvey.setUser(user);
     if (institutionalAffiliations != null) {
       final User u = user;
-      institutionalAffiliations.forEach(
-          affiliation -> {
-            affiliation.setUser(u);
-            u.addInstitutionalAffiliation(affiliation);
-          });
+      institutionalAffiliations.stream()
+          .filter(Objects::nonNull)
```

Comment:
In my view, the bug is that currently the client code is sending:

```
/createAccount {
  institutionalAffiliations: [{
    institution: undefined,
    role: undefined
  }]
}
```

client side I would either filter that out when calling `createAccount`, or maybe just always filter it out if the flag is off.